File: inventory-management/src/main/java/com/inventory_management/service/UserService.java
package com.inventory_management.service;

import com.inventory_management.dto.request.UserRequestDTO;
import com.inventory_management.dto.response.UserResponseDTO;
import org.springframework.data.domain.Page;

public interface UserService {

    UserResponseDTO createUser(UserRequestDTO request);

    Page<UserResponseDTO> getAllUsers(
            Boolean active,
            int page,
            int size,
            String sortBy,
            String sortDir
    );

    Page<UserResponseDTO> searchUsers(
            String keyword,
            Boolean active,
            int page,
            int size,
            String sortBy,
            String sortDir
    );

    UserResponseDTO getUserById(Integer userId);

    UserResponseDTO updateUser(
            Integer userId,
            UserRequestDTO request
    );

    void deleteUser(Integer userId);

    void restoreUser(Integer userId);

}

File: inventory-management-frontend/src/services/userService.js
import axiosInstance from "../api/axiosConfig";

const userService = {

    async getActiveUsers(params){

        const response = await axiosInstance.get(
            "/users",
            {
                params: {
                    ...params,
                    active: true
                }
            }
        );

        return response.data;
    },



    async getInactiveUsers(params){

        const response = await axiosInstance.get(
            "/users",
            {
                params: {
                    ...params,
                    active: false
                }
            }
        );

        return response.data;
    },



    async searchUsers(params){

        const response = await axiosInstance.get(
            "/users/search",
            {
                params
            }
        );

        return response.data;
    },



    async createUser(data){

        const response = await axiosInstance.post(
            "/users",
            data
        );

        return response.data;
    },



    async updateUser(id, data){

        const response = await axiosInstance.put(
            `/users/${id}`,
            data
        );

        return response.data;
    },



    async deleteUser(id){

        await axiosInstance.delete(
            `/users/${id}`
        );

    },



    async restoreUser(id){

        await axiosInstance.patch(
            `/users/${id}/restore`
        );

    }

};

export default userService;